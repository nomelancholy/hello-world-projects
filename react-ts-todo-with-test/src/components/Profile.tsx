import React, { useState } from "react";

interface Props {
  userName: string;
  name: string;
}

const Profile = ({ userName, name }: Props) => {

    const [inputText, setInputText] = useState('')

 const handleChange = (e:any) => {
     setInputText(e.target.value)
 }


  return (
    <div>
        <input type="text" onChange={handleChange} value={inputText} placeholder="anything"/>
      <span>
        <b>{userName}</b>
      </span>
      <span>{name}</span>
    </div>
  );
};

export default Profile;
