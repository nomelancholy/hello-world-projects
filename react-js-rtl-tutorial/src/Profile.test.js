import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="Take Knowledge" name="최재욱" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile username="Take Knowledge" name="최재욱" />);
    utils.getByText('Take Knowledge'); // Take Knowledge 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('(최재욱)'); // (최재욱) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/최/); // 정규식 /최/ 를 통과하는 엘리먼트가 있는지 확인
  });
});