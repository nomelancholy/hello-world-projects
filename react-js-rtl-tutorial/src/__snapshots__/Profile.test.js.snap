// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      Take Knowledge
    </b>
     
    <span>
      (
      최재욱
      )
    </span>
  </div>
</div>
`;
